### ISSUE TITLE
Problem with CompileSW

### ISSUE THREAD

**Issue by LegendOfDragoon** [project64/project64#926]
I get a sync error in Fushigi no Dungeon because it is not calling PI_DMA_WRITE because Opcode.base is not const.

Code at Last Sync PC:
80029BF4: lui	a0, 0xA460
80029BF8: ori	a0, a0, 0x4
80029BFC: lui	v1, 0xA460
80029C00: sw	v0, 0x0 (v1)
80029C04: lw	v0, 0xC (s0)
80029C08: lui	v1, 0x1FFF
80029C0C: ori	v1, v1, 0xFFFF
80029C10: or	v0, v0, s2
80029C14: and	v0, v0, v1
80029C18: sw	v0, 0x0 (a0)
80029C1C: beqz	s1, 0x80029C34
80029C20: addiu	v0, r0, 0x1
80029C24: beq	s1, v0, 0x80029C40
80029C28: lui	v1, 0xA460
80029C2C: j	0x80029C50
80029C30: addiu	v0, r0, 0xFFFF
80029C34: lui	v1, 0xA460
80029C38: j	0x80029C44
80029C3C: ori	v1, v1, 0xC
80029C40: ori	v1, v1, 0x8
80029C44: addiu	v0, s3, 0xFFFF
80029C48: sw	v0, 0x0 (v1)

The problem is at 80029C48, because v1 could either be 0xA4600008 or 0xA460000C.

**Comment by AmbientMalice** [comment]
My knowledge of this subject is based entirely on reading the N64 programming manual, but Indiana Jones has a PI_DMA_WRITE (to ROM) error when it crashes at the end of each level. Any possible connection?

In this case, that aside, is the game trying to do DMA writes to SRAM?

**Comment by project64** [comment]
it should still write to the rdram, which should cause an exception and the exception handler should generate the call to PI_DMA_WRITE

**Comment by LegendOfDragoon** [comment]
>it should still write to the rdram, which should cause an exception and the exception handler should generate the call to PI_DMA_WRITE

You're right, it does. I guess I should look at the exception handling code more. I got the sync error because I set the sra file to read only, because I was going to look into that stuttering issue.

>Indiana Jones has a PI_DMA_WRITE (to ROM) error when it crashes at the end of each level. Any possible connection?

I actually never beat the first level. I don't really know how to play the game. I can try looking into it though.

>In this case, that aside, is the game trying to do DMA writes to SRAM?

I never got far since this sync error I got, happened soon after booting. I do want to see if I can find a good solution to the constant saving issue. Not sure whether I will, but it'll be interesting to try and solve.
